ResearchKit/Onboarding/ORKRegistrationStep.m

```diff
@@ -0,0 +1,258 @@
+        answerFormat.spellCheckingType = UITextSpellCheckingTypeNo;
+
+        ORKFormItem *passwordItem = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemPassword
+                                                               text:ORKLocalizedString(@"PASSWORD_FORM_ITEM_TITLE", nil)
+                                                       answerFormat:answerFormat];
+        passwordItem.placeholder = ORKLocalizedString(@"PASSWORD_FORM_ITEM_PLACEHOLDER", nil);
+        passwordItem.optional = NO;
+        
+        [formItems addObject:passwordItem];
+        
+        NSString *originalItemIdentifier = [passwordItem.identifier copy];
+        ORKConfirmTextAnswerFormat *confirmAnswerFormat = [[ORKConfirmTextAnswerFormat alloc]
+                                                           initWithOriginalItemIdentifier:originalItemIdentifier];
+        confirmAnswerFormat.multipleLines = NO;
+        confirmAnswerFormat.secureTextEntry = YES;
+        confirmAnswerFormat.spellCheckingType = UITextSpellCheckingTypeNo;
+        confirmAnswerFormat.autocapitalizationType = UITextAutocapitalizationTypeNone;
+        confirmAnswerFormat.autocorrectionType = UITextAutocorrectionTypeNo;
+
+        ORKFormItem *confirmItem = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemConfirmPassword
+                                                                      text:ORKLocalizedString(@"CONFIRM_PASSWORD_FORM_ITEM_TITLE", nil)
+                                                               answerFormat:confirmAnswerFormat];
+        confirmItem.placeholder = ORKLocalizedString(@"CONFIRM_PASSWORD_FORM_ITEM_PLACEHOLDER", nil);
+        confirmItem.optional = NO;
+        
+        [formItems addObject:confirmItem];
+    }
+
+    if (!(_options & ORKRegistrationStepDefault)) {
+        ORKFormItem *item = [[ORKFormItem alloc] initWithSectionTitle:ORKLocalizedString(@"ADDITIONAL_INFO_SECTION_TITLE", nil)];
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGivenName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemGivenName
+                                                               text:ORKLocalizedString(@"CONSENT_NAME_GIVEN", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"GIVEN_NAME_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeFamilyName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemFamilyName
```

Comment:
`FamilyName` and `GivenName` is a little bit tricky:) 
Please refer to the consent `FamilyName` and `GivenName` section to make sure you list them in the right order.